device/common/include/traccc/seeding/device/select_seeds.hpp

```diff
@@ -0,0 +1,43 @@
+// Project include(s).
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/device/get_prefix_sum.hpp"
+#include "traccc/edm/device/doublet_counter.hpp"
+#include "traccc/edm/seed.hpp"
+#include "traccc/seeding/detail/seeding_config.hpp"
+#include "traccc/seeding/detail/spacepoint_grid.hpp"
+#include "traccc/seeding/detail/triplet.hpp"
+
+// System include(s).
+#include <cassert>
+#include <cstddef>
+
+namespace traccc::device {
+
+/// Function used for selecting good triplets to be recorded into seed container
+///
+/// @param[in] globalIndex   The index of the current thread
+/// @param[in] filter_config seed filter config
+/// @param[in] spacepoints_view vecmem container for internal spacepoint
+/// @param[in] dc_prefix_sum view of the vector prefix sum on
+/// doublet_counter_container
+/// @param[in] doublet_counter_container vecmem container for doublet_counter
+/// @param[in] tc_view view on vecmem container for triplets
+/// @param[in] data Array for temporary storage of triplets for
+/// comparison
+/// @param[out] seed_container vecmem container for seeds
+///
+TRACCC_HOST_DEVICE
+void select_seeds(
+    const std::size_t globalIndex, const seedfilter_config& filter_config,
```

Comment:
This is absolutely just a coding style question... I believe the wide recommendation is that arguments passed by value should not be declared as `const` in the interface declaration of a function. Since the function receives a copy of the `globalIndex` variable, the caller of this function does not care if the function internally will modify this copy of the global index or not.

Usually the coding style recommendation is to omit any `const` modifiers from the header file, and only declare the variables as `const` in the implementation.

Note that this is **only** for arguments passed by value. For arguments passed by pointer or reference, const-ness is very important for the caller of the function. So there we are not allowed to declare the variable differently in the header and the implementation.